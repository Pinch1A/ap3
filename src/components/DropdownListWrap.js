import React from 'react'
import DropdownItem from './DropdownItem'
import styled from 'styled-components'

const ListWrap = styled.div`
  -webkit-box-shadow: 10px 10px 35px -15px rgba(0, 0, 0, 0.58);
  -moz-box-shadow: 10px 10px 35px -15px rgba(0, 0, 0, 0.58);
  box-shadow: 10px 10px 35px -15px rgba(0, 0, 0, 0.58);
  border: 1px solid transparent;
  border-radius: 6px;
  margin: 3px 0;
  max-height: 550px;
  overflow: auto;
`
const HasMore = styled.button`
  border: none;
  width: 100%;
  height: 40px;
  font-size: 20px;
  cursor: pointer;
  letter-spacing: 1px;
  outline: none;
`

const DropdownListWrap = ({ data, setSelected, setOpen, selected, loadMore, hasMore }) => {
  const selectCountry = (item) => {
    setSelected(item)
    setOpen(false)
  }

  return (
    <ListWrap>
      {data.map((item, index) => (
        <DropdownItem
          {...item}
          key={index}
          isSelected={selected.name === item.name}
          setSelected={selectCountry}
        />
      ))}
      {hasMore && <HasMore onClick={() => loadMore()}>Load More</HasMore>}
    </ListWrap>
  )
}
export default DropdownListWrap
