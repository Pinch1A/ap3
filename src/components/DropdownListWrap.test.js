import DropdownListWrap from './DropdownListWrap'
import React from 'react'
import { shallow, mount } from 'enzyme'

const props = {
  data: [
    { name: 'test1', flag: 'flag1' },
    { name: 'test2', flag: 'flag2' },
  ],
  setSelected: jest.fn(),
  setOpen: jest.fn(),
  selected: { name: '', flag: null },
  loadMore: jest.fn(),
  hasMore: false,
}

describe('<DropdownListWrap>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<DropdownListWrap {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it(`render ${props.data.length} items`, () => {
    const wrapper = mount(<DropdownListWrap {...props} />)
    const itemList = wrapper.find('Item')
    expect(itemList).toHaveLength(2)
  })
  it(`render with hasMore button`, () => {
    const newProps = { ...props, hasMore: true }
    const wrapper = mount(<DropdownListWrap {...newProps} />)
    const HasMore = wrapper.find('button')
    expect(HasMore).toBeDefined()
  })
  it(`render with hasMore button and trigger loadMore function`, () => {
    const newProps = { ...props, hasMore: true }
    const wrapper = mount(<DropdownListWrap {...newProps} />)
    const HasMore = wrapper.find('button')
    expect(HasMore).toBeDefined()
    HasMore.simulate('click')
    expect(props.loadMore.mock.calls.length).toEqual(1)
  })
})
