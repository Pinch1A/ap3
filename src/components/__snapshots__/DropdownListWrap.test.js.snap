// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownListWrap> renders without crashing 1`] = `
<styled.div>
  <DropdownItem
    flag="flag1"
    isSelected={false}
    key="0"
    name="test1"
    setSelected={[Function]}
  />
  <DropdownItem
    flag="flag2"
    isSelected={false}
    key="1"
    name="test2"
    setSelected={[Function]}
  />
</styled.div>
`;
